src/components/ImageUpload/ImageUploadProvider.tsx

```diff
@@ -0,0 +1,273 @@
+/*
+ * Copyright 2024 Nordeck IT + Consulting GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useWidgetApi } from '@matrix-widget-toolkit/react';
+import { TFunction } from 'i18next';
+import { PropsWithChildren, createContext, useCallback, useRef } from 'react';
+import { ErrorCode, FileRejection } from 'react-dropzone';
+import { useTranslation } from 'react-i18next';
+import { determineImageSize } from '../../lib';
+import { Size } from '../../state';
+import { PromiseSettledResult } from '../../types';
+import { SnackbarDismissAction, SnackbarProps, useSnackbar } from '../Snackbar';
+import { useMaxUploadSize } from './useMaxUploadSize';
+
+/**
+ * Maximum allowed resolution in pixels.
+ */
+const maxResolutionPixels = 32_000_000;
```

Comment:
what magic number is this? ;)

Reply:
32 Megapixels. Will add an additional comment.

Source of the value was analysis of other software…

Reply:
my question was more on the origin / why this specific value - all of that is now clear!